### PR TITLE
Remove ad slots when DFP advertising disabled

### PR DESCRIPTION
**Pull request by jbreckmckye**
This resolves an issue [in production] on the childrens' book site, where a big grey box appears above the nav:

image

Previously, we would run dfp.js on these pages, and allow GPT to strip out empty ad slots. Now, we have a dfpAdvertising flag which can be set to false, and stops DFP running entirely. One advantage of this is it means pages on the childrens' book site should never request or load the GPT script. But it means we lose GPT's native functionality for removing empty ad slots.

This change makes dfp.js strip out ad slots itself if the dfpAdvertising flag is set to false. So the ugly grey box disappears:

image

@Calanthe @uplne

### CODE REVIEW

**Comment by Calanthe** [comment]
:+1: 
I assume it will be fine but did you test it with adblock?

**Comment by jbreckmckye** [comment]
@Calanthe - I just checked, I don't see any problems.

**Comment by jbreckmckye** [comment]
@Calanthe , @uplne - something else I noticed was that the DFP 'addSlot' and 'refreshSlot' methods were still being exposed on a disabled DFP object, but could potentially throw errors, as they would make calls to a non-existing googletag object. So I added an extra commit to make those methods throw specific exceptions immediately - https://github.com/guardian/frontend/commit/15a055cb80182f4baa3e6b624a3bbe0a6c9d638f

File: static/src/javascripts/projects/common/modules/commercial/dfp.js [hunk 1]
```diff
@@ -343,6 +345,10 @@ define([
             displayed = true;
         },
         addSlot = function ($adSlot) {
+            if (!commercialFeatures.dfpAdvertising) {
+                throw new Error('DFP advertising is disabled');
```
Comment (uplne) [inline static/src/javascripts/projects/common/modules/commercial/dfp.js]: What about raven for this?

Comment (jbreckmckye) [inline static/src/javascripts/projects/common/modules/commercial/dfp.js]: I'm thinking that if there's a specific error we can anticipate, throwing a custom message with the relevant reason makes diagnosis easier.

Comment (jbreckmckye) [inline static/src/javascripts/projects/common/modules/commercial/dfp.js]: I've removed these - 0c510ed

**Pull request merged**